src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -310,6 +310,73 @@ public Response getUserPreferences(@Context final HttpServletRequest httpServlet
         }
     }
 
+    /**
+     * An endpoint for group managers (often teachers) to send a password reset request email to group members without
+     * having to know the group members account email.
+     *
+     * @param request - request information used for caching
+     * @param httpServletRequest - the request, to work ou the current user
+     * @param userIdOfInterest - userId of interest - usually a the teacher's student
+     * @return a successful response regardless of whether the email exists or an error code if there is a technical
+     *         fault
+     */
+    @POST
+    @Path("users/{user_id}/resetpassword")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @GZIP
+    public Response generatePasswordResetTokenForOtherUser(@Context final Request request,
+                                                           @Context final HttpServletRequest httpServletRequest,
+                                                           @PathParam("user_id") final Long userIdOfInterest) {
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(httpServletRequest);
+
+            RegisteredUserDTO userOfInterest = userManager.getUserDTOById(userIdOfInterest);
+            if (userOfInterest == null) {
+                throw new NoUserException("No user found with this ID.");
+            }
+
+            UserSummaryDTO userOfInterestSummaryObject = userManager.convertToUserSummaryObject(userOfInterest);
+
+            // decide if the user is allowed to view this data.
+            if (!currentUser.getId().equals(userIdOfInterest)
+                    && !userAssociationManager.hasPermission(currentUser, userOfInterestSummaryObject)) {
+                return SegueErrorResponse.getIncorrectRoleResponse();
+            }
+
+            misuseMonitor.notifyEvent(currentUser.getEmail(), PasswordResetRequestMisuseHandler.class.toString());
```

Comment:
Is it sensible to make this request count towards the teacher's own password reset limit?

Reply:
Good point, it might not be a problem in practice but I think this is an improvement.
```suggestion
            misuseMonitor.notifyEvent(currentUser.getEmail() + "_group_member_reset", PasswordResetRequestMisuseHandler.class.toString());
```